manifests/value.pp

```diff
@@ -57,7 +57,8 @@
   validate_re($type, '^\w+',
     'type parameter must not be empty but it is type => "$type"')
 
-
+  # normalize the case so that our defined() check is more consistent
+  $key_normalized = downcase($key)
```

Comment:
Wouldn't this be a surprise to a user? While the registry key is case insensitive, what if the user wanted some or all of it to be upper case? Here we would pass the lower case text down to the provider to be created as lower case